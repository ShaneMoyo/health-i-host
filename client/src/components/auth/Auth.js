import React from 'react';
import { Switch, Route, Link, Redirect, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { signin, signup } from './actions';
import Credentials from './Credentials';


function Auth({ user, signin, signup, error, location, loading }) {
  const redirect = location.state ? location.state.from : '/';


  if(user) return <Redirect to={redirect}/>;

  return (
    <section class="hero is-fullheight has-text-centered is-warning login">
        <Switch>
          <Route path="/auth/signin" component={() => (
            <div class="is-warning has-text-white">
              <br/>
              <br/>
              <br/>
              <p>Not yet registered? <Link class="button is-outlined is-small is-light" to="/auth/signup">Sign Up</Link></p>
              <Credentials action="Sign In" submit={signin} loading={loading} error={error}/>
            </div>
          )}/>
          <Route path="/auth/signup" render={() => (
            <div class="is-warning has-text-white">
              <br/>
              <br/>
              <br/>
              <p>Already have an account? <Link class="button is-outlined is-small is-light" to="/auth/signin">Sign In</Link></p>
              <Credentials action="Sign Up" submit={signup} allowName={true} loading={loading} error={error}/>
            </div>
          )}/>
        </Switch>
    </section>
  );
}

export default withRouter(connect(
  ({ auth, loading, error }) => ({
    user: auth.user,
    error,
    loading
  }),
  { signup, signin }
)(Auth));
