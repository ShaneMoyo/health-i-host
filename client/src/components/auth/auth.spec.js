import React from 'react';
import { shallow } from 'enzyme';
import Auth from './Auth';
import Credentials from './Credentials';
import configureStore from 'redux-mock-store';

describe('Auth', () => {

  const initialState = {output:100}
  const mockStore = configureStore();
  let store,container

  beforeEach(()=>{
    store = mockStore(initialState)
    container = shallow(<ConnectedHome store={store} /> )
  })

  it('should render correctly Auth component', () => {
    const component = shallow(<Auth store={store}/>);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly Credentials component', () => {
    const component = shallow(<Credentials />);

    expect(component).toMatchSnapshot();
  });
});
