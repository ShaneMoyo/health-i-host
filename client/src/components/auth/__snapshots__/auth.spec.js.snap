// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should render correctly Auth component 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`Auth should render correctly Credentials component 1`] = `
<div
  class="is-warning "
>
  <div
    class="column is-waring is-4 is-offset-4"
  >
    <div
      class="box animated fadeIn is-warning"
    >
      <form
        onSubmit={[Function]}
      >
        <h4
          class="title has-text-grey"
        >
          Login
        </h4>
        <p
          class="subtitle has-text-grey"
        >
          Please login to proceed.
        </p>
        <div
          class="field"
        >
          <div
            class="control"
          >
            <input
              class="input is-medium"
              name="email"
              placeholder="Email"
            />
          </div>
        </div>
        <div
          class="field"
        >
          <div
            class="control"
          >
            <input
              class="input is-medium"
              name="password"
              placeholder="Password"
            />
          </div>
        </div>
        <div
          class="field has-text-centered"
        >
          <div
            class="control has-text-centered"
          />
        </div>
        <button
          class="button is-outlined is-large is-info"
        />
      </form>
    </div>
  </div>
</div>
`;
